File: kleineres/find-dupes.c
#include <stdio.h>
int main(){
	long size;
	char path[2049];
	while(scanf("%ld*[ \n\t]%2048s", &size, path) == 2) {
		printf("read: %ld %2048s\n", size, path);
	}
	printf("terminated: %ld %2048s\n", size, path);
}

File: kleineres/find-dupes.py
import sys
line = sys.stdin.readline()
print("read: {}".format(line))
size,path = line.split(maxsplit=1)
while(True):
	nextline = sys.stdin.readline()
	print("read: {}".format(nextline))
	nextsize,nextpath = nextline.split(maxsplit=1)
	if (size == nextsize):
		print("{} {}".format(size, path))
	while (size == nextsize):
		print("{} {}".format(nextsize, nextpath))
		nextline = sys.stdin.readline()
		print("read: {}".format(nextline))
		nextsize,nextpath = nextline.split(maxsplit=1)
	size=nextsize
	path=nextpath
